core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -0,0 +1,135 @@
+    super(conf.numPartitions);
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    validateInputFields(pipelineConfigurer.getStageConfigurer().getInputSchema());
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(conf.parseSchema());
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    outputFields = conf.getOutputFields();
+    outputMappings = conf.getOutputFieldMappings();
+  }
+
+  @Override
+  public void groupBy(StructuredRecord record, Emitter<StructuredRecord> emitter) throws Exception {
+    String keyField = conf.getKeyField();
+    Schema keyFieldSchema = Schema.recordOf("record", Schema.Field.of(keyField, Schema.of(Schema.Type.STRING)));
+    StructuredRecord.Builder builder = StructuredRecord.builder(keyFieldSchema);
+    builder.set(keyField, record.get(keyField));
+    emitter.emit(builder.build());
+  }
+
+  @Override
+  public void aggregate(StructuredRecord groupKey, Iterator<StructuredRecord> iterator,
+                        Emitter<StructuredRecord> emitter) throws Exception {
+    if (!iterator.hasNext()) {
+      return;
+    }
+    if (outputSchema == null) {
+      initializeOutputSchema();
+    }
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    builder.set(conf.getKeyField(), groupKey.get(groupKey.getSchema().getFields().get(0).getName()));
+    while (iterator.hasNext()) {
+      StructuredRecord record = iterator.next();
+      String fieldName = record.get(conf.getFieldName());
+
+      if (outputFields.contains(fieldName)) {
+        builder.set(fieldName, record.get(conf.getFieldValue()));
+      } else {
+        if ((outputMappings.containsKey(fieldName)) && (!outputMappings.get(fieldName).
+          equals(RowDenormalizerConfig.CHECK_ALIAS))) {
+          builder.set(outputMappings.get(fieldName), record.get(conf.getFieldValue()));
+        }
+      }
+    }
+    emitter.emit(builder.build());
+  }
+
+  /**
+   * Builds the output schema, using output fields provided by user
+   */
+  private void initializeOutputSchema() {
+    List<Schema.Field> fields = new ArrayList<>();
+    // Input fields coming from source will be of String type, that's why we are building the output schema as String.
+    fields.add(Schema.Field.of(conf.getKeyField(), Schema.of(Schema.Type.STRING)));
+    for (String outputField : outputFields) {
+      fields.add(Schema.Field.of(outputField, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    }
+    outputSchema = Schema.recordOf("group.key.schema", fields);
+  }
+
+  /**
+   * @param inputSchema Validates whether the keyfield, fieldname and fieldvalue entered by user is present in input
+   *                    schema or not.
+   * @return true
+   */
+  private void validateInputFields(Schema inputSchema) {
+    if (!inputSchema.getFields().toString().contains(conf.getKeyField())) {
```

Comment:
you should not be checking if a field exists in the schema by doing .toString() on a list and checking the key field is part of that string, as you can get false positives that way.  Instead, you need to iterate through the list and check that the configured key field matches one of the field names.